src/database.c

```diff
@@ -481,15 +521,185 @@ _add_to_db(ProfMessage* message, char* type, const Jid* const from_jid, const Ji
 
     if (SQLITE_OK != sqlite3_exec(g_chatlog_database, query, NULL, 0, &err_msg)) {
         if (err_msg) {
-            log_error("SQLite error: %s", err_msg);
+            log_error("SQLite error in _add_to_db(): %s", err_msg);
             sqlite3_free(err_msg);
         } else {
-            log_error("Unknown SQLite error");
+            log_error("Unknown SQLite error in _add_to_db().");
         }
     } else {
         int inserted_rows_count = sqlite3_changes(g_chatlog_database);
         if (inserted_rows_count < 1) {
             log_error("SQLite did not insert message (rows: %d, id: %s, content: %s)", inserted_rows_count, message->id, message->plain);
         }
+
+        sqlite3_int64 last_inserted_id = sqlite3_last_insert_rowid(g_chatlog_database);
+        if (last_inserted_id < 1) {
+            log_error("Message was inserted without error, but unable to get last inserted ID (got: %d)", last_inserted_id);
+            return;
+        }
+
+        if (message->replace_id) {
+            // TODO: maybe put as a trigger
```

Comment:
Please do so.

I'd prefer having a final solution merged instead of re-iterating again and again since introducing this would bump the db version again, or not?

Reply:
Done.